fix #2565

spnettec opened the pull request:
Signed-off-by: heyoulin <heyoulin@gmail.com>

Error appears after changing console username/password

**Related Issue:** #2565

spnettec committed:
fix #2565

Signed-off-by: heyoulin <heyoulin@gmail.com>

mstankovic commented:
@MMaiero @spnettec I checked this PR and i was unable to recreate #2565
i also checked few more scenarios with logging and changing username/password and everything works fine.

spnettec committed:
fix incorrect CONSOLE_PATH

Signed-off-by: heyoulin <heyoulin@gmail.com>

spnettec commented:
@nicolatimeus  I think this bug has been fixed, please review it.

spnettec committed:
fix lambda expression 

I don't know why consoleOptions.getAppRoot()::equals does't work.  instead of p->consoleOptions.getAppRoot().equals(p)

spnettec committed:
cleanup and format

Signed-off-by: heyoulin <heyoulin@gmail.com>

spnettec committed:
fix signed by

Signed-off-by: heyoulin <heyoulin@gmail.com>

spnettec committed:
add signed by

Signed-off-by: heyoulin <heyoulin@gmail.com>

nicolatimeus commented on kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/Console.java:
```diff
@@ -372,80 +391,82 @@ private HttpContext initResourceContext(final HttpContext defaultContext) {
 
         this.httpService.registerServlet(PASSWORD_AUTH_PATH,
-                new GwtPasswordAuthenticationServiceImpl(this.authMgr, CONSOLE_PATH), null, sessionContext);
+                new GwtPasswordAuthenticationServiceImpl(this.authMgr, () -> this.appRoot), null, this.sessionContext);
```
The `appRoot` parameter was originally meant to allow having a 3rd party web ui available on the root. It is not related the console path. Maybe we can improve the parameter description to avoid misunderstanding. If you want to enable the customisation of the console path (which can be desirable) please introduce another configuration parameter.

spnettec replied:
I don't understand what means this appRoot is. If change the appRoot this code
`
this.httpService.registerServlet("/", new RedirectServlet("/"::equals, this.appRoot), null, resourceContext);
`
will redirect console to the changed appRoot. That's why i thought appRoot is console path.

By the way i don't know why Eclipse CLA check is aways failing.

spnettec replied:
Should i change to this?
`this.httpService.registerServlet("/", new RedirectServlet("/"::equals, CONSOLE_PATH), null, resourceContext);`

If not next time the we can't enter console by root path '/'.

nicolatimeus replied:
I will try to explain better the use case for `appRoot`:

Suppose you want to add another web application to the gateway that is completely different from Kura web ui, for example a web application developed by a customer that manages only its application, and the customer wants it to be available on the root path `/`.

You can achieve this in the following way:

1. Create a custom bundle that serves the application on `/mycompany/myapp`.
2. Set the `appRoot` parameter to `/mycompany/myapp`.

In this way the custom application will be available on the root path, Kura console will be still available on `/admin/console` which is currently hardcoded.

In other words `appRoot` is simply the relative path to which the user will be redirected when accessing the root path, which is not necessarily the Kura console path.

spnettec replied:
Thanks. I see.  You can review the change of #2568 about this and i'll close this PR and open other PR later because of Eclipse CLA failing

nicolatimeus reviewed (changes requested):
Hi @spnettec, the changes look good to me, just please address the comment below. Also please check why Eclipse CLA check is failing.

spnettec commented:
Close this PR because of Eclipse CLA failing
